src/types/geography.ts

```diff
@@ -0,0 +1,301 @@
+          .map(
+            ring =>
+              `[${ring
+                .map(point => `${point.longitude} ${point.latitude}`)
+                .join(",")}]`
+          )
+          .join(",")}]}`;
+      default:
+        return '{"type":"GeometryCollection","coordinates":[]}';
+    }
+  }
+
+  /**
+   * Return new instance of this Geography object
+   * @returns Geography
+   */
+  toGeoPoint(): Geography {
+    return new Geography(this.toString());
+  }
+
+  /**
+   * Return new instance of this Geography object
+   * @returns Geography
+   */
+  toGeoLine(): Geography {
+    return new Geography(this.toString());
+  }
+
+  /**
+   * Return new instance of this Geography object
+   * @returns Geography
+   */
+  toGeoPolygon(): Geography {
+    return new Geography(this.toString());
+  }
+
+  /**
+   * Load Geography object represented by the WKT text
+   */
+  fromString(str: string) {
+    this.parse(str);
+  }
+
+  /**
+   * Return the distance (in meters) between the current point and the parameter
+   * @returns number
+   */
+  distance(geo: Geography): number {
+    return Geography.distance(this, geo);
+  }
+
+  /**
+   * Returns true if the Geographic object parameter includes or intersects the current object
+   * @returns boolean
+   */
+  intersect(geo: Geography): boolean {
+    return Geography.intersect(this, geo);
+  }
+
+  /**
+   * Create Geography object represented by the WKT text
+   * @returns Geography
+   */
+  static fromString(str: string): Geography {
+    return new Geography(str);
+  }
+
+  /**
+   * Return the distance (in meters) between points
+   * @returns number
+   */
+  static distance(from: Geography, to: Geography): number {
+    // Haversine formula:
+    // a = sin(delta_lat/2) + cos(lat1).cos(lat2).sin(delta_long/2)
+    // c = 2.atan2(sqrt(a), sqrt(1-a))
+    // d = R.c
+
+    if (from.isEmpty() || to.isEmpty()) return 0;
+
+    const earthRadius = 6371;
+
+    const deltaLatitude = (to.latitude - from.latitude) * (Math.PI / 180);
+    const deltaLongitude = (to.longitude - from.longitude) * (Math.PI / 180);
+
+    const a1 = Math.pow(Math.sin(deltaLatitude / 2), 2);
+    const a2 = Math.pow(Math.sin(deltaLongitude / 2), 2);
+    const a3 =
+      Math.cos(from.latitude * (Math.PI / 180)) *
+      Math.cos(to.latitude * (Math.PI / 180));
+    const a = a1 + a2 * a3;
+
+    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+
+    return Math.round(earthRadius * c * 1000); // distance in meters
+  }
+
+  /**
+   * Returns true if the two Geographic objects instersect.
+   * @returns boolean
+   */
+  static intersect(a: Geography, b: Geography): boolean {
+    // TODO: not implemented
+    return false;
+  }
+
+  private parse(str: string) {
+    const coordsTestRegex = /^(-?\d*(?:\.\d+)?)(?:\s*,\s*| +)(-?\d*(?:\.\d+)?)$/i; // latitude,longitude
+    const pointTestRegex = /POINT\s*\(\s*(-?\d*(?:\.\d+)?) +(-?\d*(?:\.\d+)?)\s*\)/i; // POINT(longitude latitude)
```

Comment:
This regex is repeated in line 249. Moving it outside the class, to a const would be great!